hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/AbstractSessionCredentialsProvider.java

```diff
@@ -70,21 +73,27 @@ public AbstractSessionCredentialsProvider(
   /**
    * Initialize the credentials by calling
    * {@link #createCredentials(Configuration)} with the current config.
-   * @throws IOException on any failure.
+   * Sets {@link AbstractSessionCredentialsProvider#initializationException} on failure
    */
   @Retries.OnceTranslated
-  protected void init() throws IOException {
+  protected void init() {
     // stop re-entrant attempts
     if (initialized.getAndSet(true)) {
       return;
     }
-    try {
-      awsCredentials = Invoker.once("create credentials", "",
-          () -> createCredentials(getConf()));
-    } catch (IOException e) {
-      initializationException = e;
-      throw e;
-    }
+    Runnable initTask = () -> {
+      try {
+        AWSCredentials credentials = Invoker.once(
+            "create credentials", "", () -> createCredentials(getConf()));
+        awsCredentials.complete(credentials);
+      } catch (IOException e) {
+        awsCredentials.completeExceptionally(e);
+      }
+    };
+
+    awsCredentials.handle((result, t) -> initializationException = (IOException) t);
+
+    ForkJoinPool.commonPool().submit(initTask);
```

Comment:
this is overkill for what, in the implementations in our own source tree, are just local lookups